src/Controllers/RedirectController.php

```diff
@@ -2,22 +2,57 @@
 
 namespace DT\Launcher\Controllers;
 
+
+use Disciple_Tools_Users;
 use DT\Launcher\Illuminate\Http\Request;
 use DT\Launcher\Illuminate\Http\Response;
+use DT_Magic_URL;
 use function DT\Launcher\magic_url;
 use function DT\Launcher\redirect;
 
-class RedirectController {
-
-	/**
-	 * Redirects to the URL for the launcher app. This uses the
-	 * auth middleware, so the user will be redirected to
-	 * the login page if they are not logged in.
-	 *
-	 * @param Request $request The HTTP request object.
-	 * @param Response $response The HTTP response object.
-	 */
-	public function show( Request $request, Response $response ) {
-		return redirect( magic_url() );
-	}
+class RedirectController
+{
+
+    /**
+     * Redirects to the URL for the launcher app. This uses the
+     * auth middleware, so the user will be redirected to
+     * the login page if they are not logged in.
+     *
+     * @param Request $request The HTTP request object.
+     * @param Response $response The HTTP response object.
+     */
+    public function show(Request $request, Response $response)
+    {
+
+        global $wpdb;
+
+        $preference_key = 'dt-launcher-app';
+        $meta_key = $wpdb->prefix . DT_Magic_URL::get_public_key_meta_key('dt-launcher', 'app');
```

Comment:
The first parameter of get_public_key_meta_key should be "launcher" to match the root set in the launcher magic link.